components/support/utils/src/main/java/mozilla/components/support/utils/URLStringUtils.kt

```diff
@@ -28,4 +28,10 @@ object URLStringUtils {
         }
         return uri.toString()
     }
+
+    private val isURLLenient by lazy {
+        // Be lenient about what is classified as potentially a URL. Anything that contains a :
+        // or . and has no internal spaces is potentially a URL.
+        "^\\s*\\w+(://|:|\\.)\\w+\\S*\\s*$".toRegex()
```

Comment:
Why is `\\w+\\S*` necessary? It seems redundant.

Reply:
We need both the \w+ and \S* because they seem to match slightly different things. The \S matches any non-whitespace character (e.g., '~') and \w matches only word characters. In other words, the regex is requiring that there be a non-symbol character somewhere after the ., : or :// and before any other character or the end of the string. 

For example, we want to match something like mozilla.com/~userdir but not something like mozilla./~ or mozilla:/

Reply:
I see: I think we should we include a comment to explain that.

From my perspective, I'm not sure being so strict is necessary but I haven't done the research. :)